lib/compat/wordpress-6.2/class-gutenberg-rest-global-styles-controller-6-2.php

```diff
@@ -201,4 +201,54 @@ private function validate_custom_css( $css ) {
 		}
 		return true;
 	}
+
+	/**
+	 * Returns the given theme global styles config.
+	 * Duplicated from core. The only change is that we call WP_Theme_JSON_Resolver_6_2::get_merged_data( 'theme' ) instead of WP_Theme_JSON_Resolver::get_merged_data( 'theme' ).
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param WP_REST_Request $request The request instance.
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function get_theme_item( $request ) {
+		if ( get_stylesheet() !== $request['stylesheet'] ) {
+			// This endpoint only supports the active theme for now.
+			return new WP_Error(
+				'rest_theme_not_found',
+				__( 'Theme not found.' ),
+				array( 'status' => 404 )
+			);
+		}
+
+		$theme  = WP_Theme_JSON_Resolver_6_2::get_merged_data( 'theme' );
```

Comment:
This function is duplicated so that we can fetch the updated core theme.json file. We need a better way to overload the core resolver so that we can change where it fetches from without duplicating all this code. We have to do the same thing with `get_core_data` too. cc @oandregal 

This is outside of the scope of this PR of course :)

Reply:
I see. Every time we add something to the `theme.json` we'll need to do this for it to work until the minimum WordPress version. It's cumbersome. I don't have any advice at the moment other than doing this. Perhaps we could look at making theme.json classes work differently: let core load them only if Gutenberg is not active, for example. That way we only need to have them in Gutenberg and will be used in all core places.

Reply:
Since the REST controller was removed at https://github.com/WordPress/gutenberg/pull/46102 (Gutenberg 14.8), the changes to the `theme.json` bundled in Gutenberg are not loading for the site editor (reported by @madhusudhand privately). This file/change fixes it. This could be fast-tracked to land in a separate PR.

Reply:
Moved the change to a separate PR #46810